### PR TITLE
callback error out aws_hash_table_foreach

### PR DESCRIPTION
**Pull request by TingDaoK**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/hash_table.c [hunk 1]
```diff
@@ -222,7 +222,7 @@ int aws_hash_table_init(
     AWS_PRECONDITION(alloc != NULL);
     AWS_PRECONDITION(hash_fn != NULL);
     AWS_PRECONDITION(equals_fn != NULL);
-
+    map->allocator = alloc;
```
Comment (justinboswell) [inline source/hash_table.c:225]: You're storing the allocator here, but surely it needs to be used whenever the table is re-sized right?

**Review by justinboswell — CHANGES_REQUESTED** [review]

File: source/hash_table.c (hunk 1 above)
Comment (TingDaoK) [reply source/hash_table.c:225]: Yeah, I was wanting to expose the allocator as what we did for other structures(like aws_byte_buffer). Now I see it's awkward to expose this as we have the underlying structure... Will remove.eiifccugt

File: include/aws/common/hash_table.h [hunk 2]
```diff
@@ -321,6 +322,10 @@ int aws_hash_table_remove_element(struct aws_hash_table *map, struct aws_hash_el
  *     element (if not set, iteration stops)
  * # AWS_COMMON_HASH_TABLE_ITER_DELETE   - Deletes the current value and
  *     continues iteration.  destroy_fn will NOT be invoked.
+ * # AWS_COMMON_HASH_TABLE_ITER_ERROR   - Error out and stop iteration not
```
Comment (bretambrose) [inline include/aws/common/hash_table.h:325]: This is very hard to understand grammar wise.

Comment (TingDaoK) [reply include/aws/common/hash_table.h]: Sorry about it... Updated the comment to be more grammar-correct.

**Review by justinboswell — APPROVED** [review]

**Pull request merged**